cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/model/optimized/сonsentStatus/GdprCS.kt

```diff
@@ -0,0 +1,189 @@
+            googleConsentMode = core.gcmStatus?.let { gcm -> GoogleConsentMode.initFrom(gcm) },
+            grants = core.grants.mapValues {
+                GDPRPurposeGrants(
+                    granted = it.value.vendorGrant,
+                    purposeGrants = it.value.purposeGrants
+                )
+            },
+            TCData = core.tcData
+        )
+    }
+
+    fun copyingFrom(core: ChoiceAllResponse.GDPR?, applies: Boolean?): GdprCS {
+        if (core == null) { return this }
+
+        return copy(
+            applies = applies,
+            addtlConsent = core.addtlConsent,
+            euconsent = core.euconsent,
+            dateCreated = core.dateCreated,
+            expirationDate = core.expirationDate,
+            TCData = core.tcData,
+            consentStatus = ConsentStatus.initFrom(
+                core.consentStatus
+            ),
+            postPayload = PostPayload.initFrom(core.postPayload),
+            categories = core.acceptedCategories,
+            legIntCategories = core.acceptedLegIntCategories,
+            legIntVendors = core.acceptedLegIntVendors,
+            specialFeatures = core.acceptedSpecialFeatures,
+            vendors = core.acceptedVendors,
+            webConsentPayload = core.webConsentPayload?.let { JsonConverterImpl().toJsonObject(it) },
+            googleConsentMode = core.gcmStatus?.let { gcm -> GoogleConsentMode.initFrom(gcm) },
+            grants = core.grants.mapValues {
+                GDPRPurposeGrants(
+                    granted = it.value.vendorGrant,
+                    purposeGrants = it.value.purposeGrants
+                )
+            },
+
+        )
+    }
+
+    fun copyingFrom(core: GDPRChoiceResponse): GdprCS {
+        return copy(
+            uuid = core.uuid,
+            euconsent = core.euconsent,
+            dateCreated = core.dateCreated,
+            expirationDate = core.expirationDate,
+            TCData = core.tcData,
+            consentStatus = ConsentStatus.initFrom(
+                core.consentStatus
+            ),
+            categories = core.acceptedCategories,
+            legIntCategories = core.acceptedLegIntCategories,
+            legIntVendors = core.acceptedLegIntVendors,
+            specialFeatures = core.acceptedSpecialFeatures,
+            vendors = core.acceptedVendors,
+            webConsentPayload = core.webConsentPayload?.let { JsonConverterImpl().toJsonObject(it) },
+            googleConsentMode = core.gcmStatus?.let { gcm -> GoogleConsentMode.initFrom(gcm) },
+            grants = core.grants?.mapValues {
+                GDPRPurposeGrants(
+                    granted = it.value.vendorGrant,
+                    purposeGrants = it.value.purposeGrants
+                )
+            },
+
+            )
+    }
+
+    internal fun toGDPRUserConsent(): GDPRConsentInternal {
+        return GDPRConsentInternal(
```

Comment:
Here as well, when all a function does is initialize an object, we can simply write it:
```kotlin
internal fun toGDPRUserConsent() = GDPRConsentInternal(
   ...
)
```